### ISSUE TITLE
H265ize@next

### ISSUE THREAD

**Issue by FallingSnow** [FallingSnow/h265ize#88]
### Features
- [x] Compiled with webpack and use babel.
- [ ] Have a electron-react gui

**Comment by FallingSnow** [comment]
For anyone who would like an update, progress on v6 has been steady. The backend excluding the modules has been pretty much set up and functions very well. I am now just creating the basic modules to add the original functionality back. The new version will allow anyone to create modules and swap mine out for theirs or other modules they download from [npm](https://www.npmjs.com/).

Something to keep in mind is that "h265ize-backend" does not provide a cli or gui. That will be provided by h265ize-cli and h265ize-gui respectively which will be connected to h265ize-backend via node.

Basically the system will be broken up into 4 main parts:

- h265ize-backend
- h265ize-cli
- h265ize-gui
- h265ize-module-*

Another cool feature is that h265ize will no longer be limited to any codecs. The modules will decide what codecs to encode with. However I still strive to make h265ize a fire and forget weapon, therefore defaults will be provided for everything. You should literally be able to write `h265ize video.mp4` and it will work.

I am also considering rebranding h265ize because it will no longer only serve h265, lemme know what you think.

**Comment by FallingSnow** [comment]
Another update for anyone who is reading. h265ize-backend is complete and is stable enough to be considered a beta. h265ize-cli has also made a lot of progress as well. It is able to process several files successfully but still lacks some functionality of version 4, this is my primary focus at the moment.

The majority of h265ize modules used by the backend are also complete, they include:
* encoder
* he-audio
* normalize
* autocrop
* stats
* sample
* preview-stream (Currently of no use till h265ize-gui is started.)

Modules to still be written include:
* Audio normalization
* Upconvert
* Screenshots
* File hash (Adding a crc hash to the file name is popular)
* Mulitpass (this may never be done)

**Comment by charlas** [comment]
Looking forward to it :)

**Comment by yfaykya** [comment]
How is it going? :-)

**Comment by FallingSnow** [comment]
https://github.com/NMMES/nmmes-backend

**Comment by FallingSnow** [comment]
For anyone watching... nmmes-cli is ready for alpha testing.
https://www.npmjs.com/package/nmmes-cli
